coffee_core/src/coffee.rs

```diff
@@ -261,12 +287,18 @@ impl PluginManager for CoffeeManager {
         if let Some(index) = plugins.iter().position(|x| x.name() == plugin) {
             let plugin = plugins[index].clone();
             let exec_path = plugin.exec_path.clone();
+            let root_path = plugin.root_path.clone();
+            let cloned_repositories_path = format!("{}/repositories", self.config.root_path,);
+            // make sure that we are not deleting the cloned repositories
+            if !root_path.contains(&cloned_repositories_path) {
+                fs::remove_dir_all(root_path).await?;
+            }
```

Comment:
This is not enough, if we are in the repository this means that we are running an old version of coffee and we should also remove the plugin from the configuration with the old path.

We should change the following code to work in this if else and remove the correct path

```
            self.coffee_cln_config
                .rm_conf("plugin", Some(&exec_path.to_owned()))
                .map_err(|err| CoffeeError::new(1, &err.cause))?;
                .map_err(|err| error!("{}", &err.cause))?;
```

Reply:
I can't fully understand the problem here
There are 2 cases:
- we are running the newest version of coffee --> no problem here
- we are running an older version of coffee:
      -  we only need to avoid removing the `root_path` of the plugin because that would be the cloned version.
      -  there is no problem with `exec_path` really because this path will be the same to the one we added to `cln_config` before. coffee doesn't change `exec_path` at any point in time. the new update only modifies it before adding it to `cln_config` normally. 
      
Am I missing something here?

Reply:
Also, it would be hard to craft the `exec_path` of a plugin instead of just using it from the field directly.
it would look something like `.coffee/repositories/<repo_name>/<plugin_name>` but we don't store the name of the repository from which the plugin was installed. (I think we should in the future)

Reply:
Ah my bad hahahhahha

```
[2023-07-31T05:27:35Z DEBUG coffee_core::coffee] coffee cln config: # core lightning configuration generated by coffe please do not edit this
    #plugin=/home/vincent/.coffee/repositories/folgore-git/target/release/folgore_plugin
    
    
✗ Uninst
```

look at the `#`, sorry I was sleeping yesterday,